contracts/GenArt721CoreV3_Engine.sol

```diff
@@ -0,0 +1,2137 @@
+        // packed uint: max of 100, max uint8 = 255
+        uint8 additionalPayeeSecondarySalesPercentage;
+        address payable artistAddress;
+        // packed uint: max of 100, max uint8 = 255
+        uint8 additionalPayeePrimarySalesPercentage;
+    }
+    // Project financials mapping
+    mapping(uint256 => ProjectFinance) projectIdToFinancials;
+
+    /// hash of artist's proposed payment updates to be approved by admin
+    mapping(uint256 => bytes32) public proposedArtistAddressesAndSplitsHash;
+
+    /// The render provider payment address for all primary sales revenues
+    /// (packed)
+    address payable public renderProviderPrimarySalesAddress;
+    /// Percentage of primary sales revenue allocated to the render provider
+    /// (packed)
+    // packed uint: max of 100, max uint8 = 255
+    uint8 private _renderProviderPrimarySalesPercentage = 10;
+    /// The platform provider payment address for all primary sales revenues
+    /// (packed)
+    address payable public platformProviderPrimarySalesAddress;
+    /// Percentage of primary sales revenue allocated to the platform provider
+    /// (packed)
+    // packed uint: max of 100, max uint8 = 255
+    uint8 private _platformProviderPrimarySalesPercentage = 10;
+
+    /// The render provider payment address for all secondary sales royalty
+    /// revenues
+    address payable public renderProviderSecondarySalesAddress;
+    /// Basis Points of secondary sales royalties allocated to the
+    /// render provider
+    uint256 public renderProviderSecondarySalesBPS = 250;
+    /// The platform provider payment address for all secondary sales royalty
+    /// revenues
+    address payable public platformProviderSecondarySalesAddress;
+    /// Basis Points of secondary sales royalties allocated to the
+    /// platform provider
+    uint256 public platformProviderSecondarySalesBPS = 250;
+
+    /// single minter allowed for this core contract
+    address public minterContract;
+
+    /// starting (initial) project ID on this contract
+    uint256 public immutable startingProjectId;
+
+    /// next project ID to be created
+    uint248 private _nextProjectId;
+
+    /// bool indicating if adding new projects is forbidden;
+    /// default behavior is to allow new projects
+    bool public newProjectsForbidden;
+
+    /// configuration variable (determined at time of deployment)
+    /// that determines whether or not admin approval^ should be required
+    /// to accept artist address change proposals, or if these proposals
+    /// should always auto-approve, as determined by the business process
+    /// requirements of the Engine partner using this contract.
+    ///
+    /// ^does not apply in the case where contract-ownership itself is revoked
+    bool public immutable autoApproveArtistSplitProposals;
+
+    /// version & type of this core contract
+    string public constant coreVersion = "v3.0.0";
+    string public constant coreType = "GenArt721CoreV3";
```

Comment:
Because this conforms to a new interface that indexing will have to adjust to, I think we should update this to be the following:
```suggestion
    string public constant coreVersion = "v3.1.0";
    string public constant coreType = "GenArt721CoreV3_Engine";
```

Reply:
Great point–applied.